libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp

```diff
@@ -433,6 +433,49 @@ class tags_api_impl
         return result;
     }
 
+    posts_crefs exclude(const posts_crefs& filtering_posts,
+                        const discussion_query& query,
+                        const std::function<bool(const tag_object&)>& tag_filter) const
+    {
+        auto rng_exclude = query.exclude_tags | boost::adaptors::transformed(utils::to_lower_copy)
+            | boost::adaptors::transformed([](const std::string& s) { return utils::substring(s, 0, TAG_LENGTH_MAX); });
+
+        std::set<std::string> tags_exclude(rng_exclude.begin(), rng_exclude.end());
+        if (tags_exclude.empty())
+            return filtering_posts;
+
+        std::vector<posts_crefs> exclude_posts_by_tags;
+        exclude_posts_by_tags.reserve(tags_exclude.size());
+
+        boost::transform(tags_exclude, std::back_inserter(exclude_posts_by_tags),
+                         [&](const std::string& t) { return get_posts(t, tag_filter); });
+
+        if (exclude_posts_by_tags.empty())
```

Comment:
No need to check `exclude_posts_by_tags` for emptyness

Reply:
To prevent  iterations in std::copy_if